### PR TITLE
saturn: Harmonize icon size

### PR DESCRIPTION
**Pull request by samueldr**
Not sure if it's any good (not that this is new for my contributions to saturn).

This hardcodes a specific icon size to use, *and makes icons that size*.

<details>
<summary>Screenshot</summary>

image

</details>

In the screenshot, while not obvious here, the *WrappedNeovim* would normally be huge, and the *GKrellM* icon would be small.

### CODE REVIEW

File: pkgs/saturn/main.fnl [hunk 1]
```diff
@@ -86,14 +89,18 @@
...
-      (Gtk.Image.new_from_file name)
-      (let [sizes (icon-theme:get_icon_sizes name)]
-
-        (each [_ res (pairs [64 48]) :until found]
-          (set found (icon-theme:load_icon
-                      name res
-                      (+ Gtk.IconLookupFlags.FORCE_SVG Gtk.IconLookupFlags.USE_BUILTIN))))
-        (Gtk.Image.new_from_pixbuf found))))
+    ;; UNTESTED
+    (set found (Gtk.Image.new_from_file_at_size name ICON_SIZE ICON_SIZE))
```
Comment (samueldr) [inline pkgs/saturn/main.fnl:93]: This *really needs to be tested*. Unsure what the best way is...

File: pkgs/saturn/main.fnl [hunk 2]
```diff
@@ -86,14 +89,18 @@
...
+    ;; From icon theme
+    (let [sizes (icon-theme:get_icon_sizes name)]
+      ;; Uses a list of "safe fallback" values
+      ;; Try the desired size first
+      (each [_ res (pairs [ICON_SIZE 128 64 48]) :until found]
+        (set found (icon-theme:load_icon
+                     name res
+                     (+ Gtk.IconLookupFlags.FORCE_SVG Gtk.IconLookupFlags.USE_BUILTIN))))
+      (set found (found:scale_simple ICON_SIZE ICON_SIZE GdkPixbuf.InterpType.BILINEAR))
```
Comment (samueldr) [inline pkgs/saturn/main.fnl:102]: ¯\\\_(ツ)\_/¯

Is that idiomatically fennel?

GTK ref:

 - https://docs.gtk.org/gdk-pixbuf/method.Pixbuf.scale_simple.html

Comment (telent) [reply pkgs/saturn/main.fnl:102]: Insofar as Fennel is "inspired by Clojure", probably we should aim for a functional style (I'm not especially happy about that `set` in the first place). On the other hand, mutation is pretty idiomatic for Lua  ...

I would use the threading macro here: something like (untested)

```
(set found
     (->  (icon-theme:load_icon
	   name res
	   (+ Gtk.IconLookupFlags.FORCE_SVG Gtk.IconLookupFlags.USE_BUILTIN))
       (: :scale_simple ICON_SIZE ICON_SIZE GdkPixbuf.InterpType.BILINEAR)))
```

https://fennel-lang.org/reference#and---threading-macros

Comment (telent) [reply pkgs/saturn/main.fnl:102]: indentation came out really wonky there, I think github ate some of it

Comment (samueldr) [reply pkgs/saturn/main.fnl:102]: Yep, when I went to sleep, not too long after sending this PR, I suddenly thought "oh, that threading macro would probably have worked well there".

File: pkgs/saturn/main.fnl [hunk 3]
```diff
@@ -85,15 +91,21 @@
...
-      (let [sizes (icon-theme:get_icon_sizes name)]
-
-        (each [_ res (pairs [64 48]) :until found]
-          (set found (icon-theme:load_icon
-                      name res
-                      (+ Gtk.IconLookupFlags.FORCE_SVG Gtk.IconLookupFlags.USE_BUILTIN))))
-        (Gtk.Image.new_from_pixbuf found))))
+  (if (path.absolute? name)
+    ;; From a direct path
+    (set found (GdkPixbuf.Pixbuf.new_from_file_at_size name ICON_SIZE ICON_SIZE))
```
Comment (samueldr) [inline pkgs/saturn/main.fnl:96]: `name.sub` didn't work :). I think making helpers to better declare the intent is probably better in the end, but not sure if that's what you want.

Additionally, is it idiomatic in a fennel-like lisp to use `?` in a function name like that? I don't mind removing that if it isn't.

**Comment by samueldr** [comment]
With this recent push, changes are tested correctly for all cases. But I had to somewhat cheat to better test things out. `~/.local/share/` isn't in my `XDG_DATA_DIRS`, and I'm not sure it should be. I'm not looking into that, and will share a possible fix for this.

Anyway, once I had `~/.local/share/` apps going, I took an arbitrary one and tested

  - bogus entry
  - good icon theme key
  - absolute path

All worked as intended.

File: pkgs/saturn/main.fnl (hunk 2 above)
Comment (samueldr) [reply pkgs/saturn/main.fnl]: The *maybe* variant had to be used. Convenient!

**Review by telent — APPROVED** [review]

**Pull request merged**